### PR TITLE
add input/output type constants to std-lib

### LINKED ISSUE #2186: Add input/output type constants to std-lib

### PR DESCRIPTION
**Pull request by simonr0204**
Resolves #2186 in concordance with [InputType](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#inputtype) and [OutputType](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#outputtype)

### CODE REVIEW

**Comment by simonr0204** [comment]
Any thoughts on whether these should go into `std::constants` or `std::tx` ?

**Comment by nfurfaro** [comment]
I would say add them to std::tx.

File: sway-lib-std/src/u128.sw [hunk 1]
```diff
@@ -28,7 +28,7 @@ pub trait From {
 impl From for U128 {
     pub fn from(upper: u64, lower: u64) -> U128 {
         U128 {
-            upper, lower,
+            upper, lower,
```
Comment (otrho) [inline sway-lib-std/src/u128.sw:31]: Did the formatter do this?

**Review by otrho — APPROVED** [review]

File: sway-lib-std/src/u128.sw (hunk 1 above)
Comment (simonr0204) [reply sway-lib-std/src/u128.sw:31]: Oh, weird. I assume so. I didn't touch anything here.

File: sway-lib-std/src/u256.sw [hunk 2]
```diff
@@ -29,7 +29,7 @@ pub trait From {
 impl From for U256 {
     pub fn from(a: u64, b: u64, c: u64, d: u64) -> U256 {
         U256 {
-            a, b, c, d,
+            a, b, c, d,
```
Comment (adlerjohn) [inline sway-lib-std/src/u256.sw:32]: Revert

**Review by adlerjohn — CHANGES_REQUESTED** [review]
Revert formatting changes since they're orthogonal.

File: sway-lib-std/src/u256.sw (hunk 2 above)
Comment (simonr0204) [reply sway-lib-std/src/u256.sw]: Resolved [2812ac0](https://github.com/FuelLabs/sway/pull/2187/commits/2812ac022b478045e825643b340cdb3533fb7342)

File: sway-lib-std/src/u128.sw (hunk 1 above)
Comment (simonr0204) [reply sway-lib-std/src/u128.sw]: Resolved [2812ac0](https://github.com/FuelLabs/sway/pull/2187/commits/2812ac022b478045e825643b340cdb3533fb7342)

File: sway-lib-std/src/u128.sw [hunk 3]
```diff
@@ -250,7 +250,7 @@ impl core::ops::Subtract for U128 {
         }
 
         U128 {
-            upper, lower,
+            upper, lower,
```
Comment (otrho) [inline sway-lib-std/src/u128.sw:253]: There's still this one too.

```suggestion
            upper, lower,
```

File: sway-lib-std/src/tx.sw [hunk 4]
```diff
@@ -42,6 +42,18 @@ const TX_WITNESSES_COUNT_OFFSET = 10312;
 const TX_RECEIPTS_ROOT_OFFSET = 10320;
 const TX_SCRIPT_START_OFFSET = 10352;
 
+// Input types
+const INPUT_COIN = 0u8;
+const INPUT_CONTRACT = 1u8;
+const INPUT_MESSAGE = 2u8;
+
+// Output types
+const OUTPUT_CONTRACT = 1u8;
```
Comment (adlerjohn) [inline sway-lib-std/src/tx.sw:51]: Hmm. Shouldn't there be an `OUTPUT_COIN` too?

Comment (simonr0204) [reply sway-lib-std/src/tx.sw:51]: Ha, yes - definitely. Added

**Review by adlerjohn — APPROVED** [review]

**Review by nfurfaro — APPROVED** [review]
LGTM.

**Review by nfurfaro — APPROVED** [review]

**Pull request merged**